scripts/pylib/twister/twisterlib.py

```diff
@@ -681,16 +681,19 @@ def monitor_serial(self, ser, halt_fileno, harness):
             # is available yet.
             if serial_line:
                 sl = serial_line.decode('utf-8', 'ignore').lstrip()
-                logger.debug("DEVICE: {0}".format(sl.rstrip()))
+                try:
+                    logger.debug("DEVICE: {0}".format(sl.rstrip()))
+                    log_out_fp.write(sl)
+                except UnicodeEncodeError:
+                    continue
```

Comment:
should we just ignore those chars? continue means skip all belows
```suggestion
                    Pass
```